apps/heft/src/stages/CleanStage.ts

```diff
@@ -36,16 +37,43 @@ export interface ICleanStageOptions {
  */
 export interface ICleanStageContext extends IStageContext<CleanStageHooks, ICleanStageProperties> {}
 
+interface ICleanConfigurationJson {
+  pathsToDelete: string[];
+}
+
 export class CleanStage extends StageBase<CleanStageHooks, ICleanStageProperties, ICleanStageOptions> {
   public constructor(heftConfiguration: HeftConfiguration, loggingManager: LoggingManager) {
     super(heftConfiguration, loggingManager, CleanStageHooks);
   }
 
-  protected getDefaultStageProperties(options: ICleanStageOptions): ICleanStageProperties {
+  protected async getDefaultStagePropertiesAsync(
+    options: ICleanStageOptions
+  ): Promise<ICleanStageProperties> {
+    const cleanConfigurationFileLoader: ConfigurationFile<ICleanConfigurationJson> = new ConfigurationFile<
+      ICleanConfigurationJson
+    >(path.resolve(__dirname, '..', 'schemas', 'clean.schema.json'), {
+      jsonPathMetadata: {
+        '$.pathsToDelete.*': {
+          pathResolutionMethod: PathResolutionMethod.resolvePathRelativeToProjectRoot
+        }
+      }
+    });
+
+    let cleanConfigurationFile: ICleanConfigurationJson | undefined = undefined;
+    try {
+      cleanConfigurationFile = await cleanConfigurationFileLoader.loadConfigurationFileAsync(
+        path.resolve(this.heftConfiguration.buildFolder, '.heft', 'clean.json')
+      );
+    } catch (e) {
+      if (!FileSystem.isNotExistError(e)) {
```

Comment:
A lot of Heft features are enabled based on the existence of their config file.  In the case where the file doesn't exist, we should be careful that `loadConfigurationFileAsync()` does not do any expensive work.

Also, if we could avoid throwing+catching tons of exceptions during normal operation, that would make the "Break on exceptions" debugger feature usable. :-)

Reply:
At some level, an exception will always be thrown, unless we use the `FileSystem.exists` API, which is less efficient.